ui/src/app/pages/data/data-set/export-dataset-modal.tsx

```diff
@@ -265,20 +267,27 @@ export const ExportDatasetModal = ({
       setCreatingNewNotebook(true);
     } else {
       setCreatingNewNotebook(false);
-      setIsLoadingNotebook(true);
+      setLoadingNotebookKernel(true);
       notebooksApi()
         .getNotebookKernel(
           workspace.namespace,
           workspace.id,
           appendJupyterNotebookFileSuffix(nameWithoutSuffix)
         )
-        .then((resp) => setKernelType(resp.kernelType))
+        .then((resp) => {
+          // can compare directly for now, because they exactly match
+          if (resp.kernelType !== analysisLanguage) {
+            setAnalysisLanguage(resp.kernelType);
+            // code preview will be stale if the language changed
+            getCode(resp.kernelType);
```

Comment:
this fixes an existing bug. The following could happen previously:
* code preview is open
* user selects an existing Jupyter notebook to export to
* the existing notebook file's kernel is a different language than the currently-previewed code
* the radio button shifts to the existing notebook's language
* the code preview API is called twice: once for each language
* what is ultimately displayed in the code preview box depends on who loses the race